workos/user_management.py

```diff
@@ -58,223 +68,349 @@
 PASSWORD_RESET_DETAIL_PATH = "user_management/password_reset/{0}"
 
 
+class UsersListFilters(ListArgs, total=False):
+    email: Optional[str]
+    organization_id: Optional[str]
+
+
+class InvitationsListFilters(ListArgs, total=False):
+    email: Optional[str]
+    organization_id: Optional[str]
+
+
+class OrganizationMembershipsListFilters(ListArgs, total=False):
+    user_id: Optional[str]
+    organization_id: Optional[str]
+    # A set of statuses that's concatenated into a comma-separated string
+    statuses: Optional[str]
+
+
+class AuthenticationFactorsListFilters(ListArgs, total=False):
+    user_id: str
+
+
 class UserManagementModule(Protocol):
-    def get_user(self, user_id: str) -> dict: ...
+    _http_client: Union[SyncHTTPClient, AsyncHTTPClient]
+
+    def get_user(self, user_id: str) -> User: ...
 
     def list_users(
         self,
-        email=None,
-        organization_id=None,
-        limit=None,
-        before=None,
-        after=None,
-        order=None,
-    ) -> dict: ...
-
-    def create_user(self, user: dict) -> dict: ...
-
-    def update_user(self, user_id: str, payload: dict) -> dict: ...
+        email: Optional[str] = None,
+        organization_id: Optional[str] = None,
+        limit: int = DEFAULT_LIST_RESPONSE_LIMIT,
+        before: Optional[str] = None,
+        after: Optional[str] = None,
+        order: PaginationOrder = "desc",
+    ) -> SyncOrAsyncListResource: ...
+
+    def create_user(
+        self,
+        email: str,
+        password: Optional[str] = None,
+        password_hash: Optional[str] = None,
+        password_hash_type: Optional[PasswordHashType] = None,
+        first_name: Optional[str] = None,
+        last_name: Optional[str] = None,
+        email_verified: Optional[bool] = False,
+    ) -> User: ...
+
+    def update_user(
+        self,
+        user_id: str,
+        first_name: Optional[str] = None,
+        last_name: Optional[str] = None,
+        email_verified: Optional[bool] = False,
```

Comment:
Would this mean that by default we would always pass `False` on update? Would that end up unverifying a user?

Reply:
Yep, that's not correct. Should be `= None` as it's optional and we'll defer to API behavior.